lib/response.js

```diff
@@ -70,18 +70,43 @@ res.status = function status(code) {
  *      last: 'http://api.example.com/users?page=5'
  *    });
  *
- * @param {Object} links
+ *    res.links([{
+ *          href: 'http://api.example.com/users?page=2',
+ *          rel: 'next',
+ *          title: 'next chapter',
+ *          type: 'text/plain;charset=UTF-8'
+ *      }, {
+ *          href: 'http://api.example.com/users?page=5',
+ *          rel: 'last',
+ *          title: 'last chapter',
+ *          type: 'text/plain;charset=UTF-8'
+ *      }]);
+ *
+ *
+ * @param {Object|Array} links
  * @return {ServerResponse}
  * @public
  */
 
-res.links = function(links){
+res.links = function(links) {
   var link = this.get('Link') || '';
   if (link) link += ', ';
-  return this.set('Link', link + Object.keys(links).map(function(rel){
-    return '<' + links[rel] + '>; rel="' + rel + '"';
-  }).join(', '));
-};
+  if (Array.isArray(links)) { 
+    var attrTest = /^[a-z][a-z-*]+$/;
+    for (var i = 0; i < links.length; i++) { 
+      if (i !== 0) link += ', '; // do not prepend ', ' if it's the first link
+      link += '<' + links[i].href + '>; ';
+      delete links[i].href; // already appended
+      link += Object.keys(links[i]).filter(function (key) { return attrTest.test(key); })
+        .map(function(key) { return key + '="' + links[i][key] + '"'; }).join('; ');
```

Comment:
This (and line 106) only allow quoted-string values. Token and extended notation values should also be allowed. hreflang, for example, only allows tokens. title* only allows extended notation values. See [rfc5988](https://tools.ietf.org/html/rfc5988#section-5).